tests/codegen/abi-win64-zst.rs

```diff
@@ -0,0 +1,49 @@
+//@ compile-flags: -Z merge-functions=disabled
+
+//@ revisions: linux
+//@[linux] compile-flags: --target x86_64-unknown-linux-gnu
+//@[linux] needs-llvm-components: x86
+
+//@ revisions: windows-gnu
+//@[windows-gnu] compile-flags: --target x86_64-pc-windows-gnu
+//@[windows-gnu] needs-llvm-components: x86
+
+//@ revisions: windows-msvc
+//@[windows-msvc] compile-flags: --target x86_64-pc-windows-msvc
+//@[windows-msvc] needs-llvm-components: x86
+
+#![feature(no_core, lang_items, rustc_attrs, abi_vectorcall)]
+#![no_core]
+#![crate_type = "lib"]
+
+#[lang = "sized"]
+trait Sized {}
+
+// Make sure the argument is always ignored when explicitly requesting "win64" or "sysv64" ABI.
+
+// CHECK: define win64cc void @pass_zst_win64()
+#[no_mangle]
+extern "win64" fn pass_zst_win64(_: ()) {}
+
+// CHECK: define x86_64_sysvcc void @pass_zst_sysv64()
+#[no_mangle]
+extern "sysv64" fn pass_zst_sysv64(_: ()) {}
+
+// CHECK: define x86_vectorcallcc void @pass_zst_vectorcall()
+#[no_mangle]
+extern "vectorcall" fn pass_zst_vectorcall(_: ()) {}
+
+// windows-gnu: define x86_fastcallcc void @pass_zst_fastcall()
+// windows-msvc: define x86_fastcallcc void @pass_zst_fastcall()
```

Comment:
Previously we didn't add the `x86_fastcallcc` here since we mapped this to the "C" calling convention. Is it a problem for LLVM to have `x86_fastcallcc` set on an x86-64 build?

Reply:
I managed to avoid changing the LLVM-level calling convention here.